test/reservation_maker_test.rb

```diff
@@ -0,0 +1,63 @@
+require_relative 'test_helper'
+
+describe Hotel::Reservation_Maker do
+  before do 
+    @reservation_maker = Hotel::Reservation_Maker.new
+    @date = Date.parse("2020-08-04")
+  end 
+
+  describe "wave 1" do
+    describe "initiation" do
+      it "returns a list of all rooms" do
+        rooms = @reservation_maker.rooms
+        expect(rooms).must_be_kind_of Array
+        expect(rooms.length).must_equal 20
+      end
+    end 
+    
+    describe "reserve room" do
+      it "can take two dates and return a Reservation" do
+        check_in = @date
+        check_out = @date + 4
+        reservation = @reservation_maker.reserve_room(check_in, check_out)
+        expect(reservation).must_be_kind_of Hotel::Reservation 
+      end 
+    end 
+    
+    describe "add reservation" do
+      #Check to see that add_reservations method outputs an array of all reservations which exist
+      it "is a list of reservations" do
+        check_in = @date
+        check_out = @date + 4
+        reservation = @reservation_maker.reserve_room(check_in, check_out)
```

Comment:
To test this `add_reservation` method in isolation, you probably don't need to call `reserve_room` as part of the Arrange step, you probably can just make a new instance of `Reservation`! Let me know if that feels confusing